storage.bs

```diff
@@ -94,37 +100,292 @@ function retrieveNextChunk(nextChunkInfo) {
+infrastructure defined by this standard to keep track of its storage needs.
+
+<p>A <a>storage endpoint</a> has an <dfn for="storage endpoint">identifier</dfn>, which is a
+<a>storage identifier</a>.
+
+<p>A <a>storage endpoint</a> also has <dfn for="storage endpoint">types</dfn>, which is a
+<a for=/>set</a> of <a>storage types</a>.
+
+<p>A <dfn>storage identifier</dfn> is an <a for=/>ASCII string</a>.
+
+<p>A <dfn>storage type</dfn> is "<code>local</code>" or "<code>session</code>".
+
+<hr>
+
+<p>The <dfn>registered storage endpoints</dfn> are a <a for=/>set</a> of <a>storage endpoints</a>
+defined by the following table:
+<!-- Note, these will be exposed more generally through https://github.com/whatwg/storage/pull/69 so
+     please keep https://w3ctag.github.io/design-principles/#casing-rules in mind. -->
+
+<table>
+ <tr>
+  <th><a>Storage identifier</a>
+  <th><a>Storage types</a>
+ <tr>
+  <td>"<code>caches</code>"
+  <td>« "<code>local</code>" »
+ <tr>
+  <td>"<code>indexedDB</code>"
+  <td>« "<code>local</code>" »
+ <tr>
+  <td>"<code>localStorage</code>"
+  <td>« "<code>local</code>" »
+ <tr>
+  <td>"<code>serviceWorkerRegistrations</code>"
+  <td>« "<code>local</code>" »
+ <tr>
+  <td>"<code>sessionStorage</code>"
+  <td>« "<code>session</code>" »
+</table>
+
+<p class="note no-backref">As mentioned, standards can use these <a>storage identifiers</a> with
+<a>obtain a local storage bottle map</a> and <a>obtain a session storage bottle map</a>. It is
+anticipated that some APIs will be applicable to both <a>storage types</a> going forward.
+<!-- If that does not happen by 2024 we probably ought to admit defeat and simplify this a bit. -->
+
+
+<h3 id=storage-sheds>Storage sheds</h3>
+
+<p>A <dfn>storage key</dfn> is an <a for=/>origin</a>. [[HTML]]
+
+<p class=XXX>This is expected to change, see
+<a href="https://privacycg.github.io/storage-partitioning/">Client-Side Storage Partitioning</a>.
+
+<p>A <dfn>storage shed</dfn> is a <a for=/>map</a> of <a>storage keys</a> to <a>storage shelves</a>.
+It is initially empty.
+
+<hr>
+
+<p>A <a for=/>user agent</a> holds a <dfn for="user agent">storage shed</dfn>, which is a
+<a for=/>storage shed</a>.
+
+<p>A browsing session holds a <dfn for="browsing session">storage shed</dfn>, which is a
+<a for=/>storage shed</a>.
+
+<p class="XXX" id=lack-of-browsing-session>See
+<a href="https://github.com/whatwg/html/issues/4782">whatwg/html issue #4782</a> and
+<a href="https://github.com/whatwg/html/issues/5350">whatwg/html issue #5350</a> for defining
+browsing session. It is roughly analogous to <a>top-level browsing context</a> except that it cannot
+be replaced due to <code>Cross-Origin-Opener-Policy</code> or navigation.
+
+
+<h3 id=storage-shelves>Storage shelves</h3>
+
+<p>A <dfn oldids=site-storage-unit lt="storage shelf|storage shelves">storage shelf</dfn> is the
+biggest storage container, apart from the user agent itself. It holds a <dfn>bucket map</dfn>, which
```

Comment:
Isn't a shed bigger than a shelf? (I'm not sure of the utility of calling out "biggest storage container" here)

Reply:
I also find this bit a little odd and after the terminology change probably no longer necessary to clarify the sizes beyond the hierarchical containment relationship.

Right now it's:
> A storage shelf is the biggest storage container, apart from the user agent itself. It holds a bucket map, which is a map of strings to storage buckets. 

And maybe it could instead be something like:
> **A storage shelf is the unit of per-origin data storage within a storage shed.**  It holds a bucket map, which is a map of strings to storage buckets. 

Alternately:
> **A storage shelf exists for each origin within a storage shed.  Each storage shelf holds a bucket map that maps strings to storage buckets.**